src/ophys_etl/modules/segmentation/modules/schemas.py

```diff
@@ -150,6 +151,43 @@ class SegmentV0InputSchema(argschema.ArgSchema):
         description="if provided, will write subgraphs to json as ROIs")
 
 
+class PCADenoiseInputSchema(argschema.ArgSchema):
+    log_level = argschema.fields.LogLevel(default="INFO")
+    video_path = argschema.fields.InputFile(
+        required=True,
+        description=("path to hdf5 video with movie stored "
+                     "in dataset 'data' nframes x nrow x ncol"))
+    video_output = argschema.fields.OutputFile(
+        required=True,
+        description="destination path to filtered hdf5 video ")
+    n_components = argschema.fields.Int(
+        required=True,
+        description=("number of principal components to keep. "
+                     "the chunking of the movie for incremental "
+                     "PCA must keep the chunksize above this number."))
```

Comment:
It took me a bit to figure out what this meant. Maybe say "the chunking of the movie for incremental PCA must ensure that there are more than n_components timesteps in each chunk."